pkg/gadgets/socket-collector/gadget.go

```diff
@@ -72,71 +75,85 @@ func (t *Trace) Operation(trace *gadgetv1alpha1.Trace,
 	params map[string]string) {
 
 	if trace.ObjectMeta.Namespace != gadgets.TRACE_DEFAULT_NAMESPACE {
-		gadgets.SetStatusError(trace, fmt.Sprintf("This gadget only accepts operations on traces in the %s namespace", gadgets.TRACE_DEFAULT_NAMESPACE))
+		gadgets.SetStatusError(trace, fmt.Sprintf("This gadget only accepts operations on traces in the %s namespace",
+			gadgets.TRACE_DEFAULT_NAMESPACE))
 		return
 	}
 
 	switch operation {
 	case "start":
-		var pid uint32 = 0
-
 		if trace.Spec.Filter == nil {
 			gadgets.SetStatusError(trace, "Missing filter")
```

Comment:
Now that it's not mandatory to specify 1 pod in the filter, could we accept `trace.Spec.Filter == nil` as the filter that accepts everything. 

ContainerSelectorFromContainerFilter() seems to handle this correctly.

Reply:
That's true. It would be only possible if you deploy manually the trace instead of using the `kubectl-gadget`, though. 

I updated the PR, PTAL.